ethcore/src/state/mod.rs

```diff
@@ -689,7 +696,10 @@ impl<B: Backend> State<B> {
 
 		// account is not found in the global cache, get from the DB and insert into local
 		let from_rlp = |b: &[u8]| Account::from_rlp(b).expect("decoding db value failed");
-		let maybe_acc = self.mkvs.get(&address).map(|value| from_rlp(&value));
+		let maybe_acc = self
+			.mkvs
+			.get(&get_metadata_key(address))
```

Comment:
this is super trivial and you should definitely feel free to ignore it, but I think you can do
```rust
.get(&get_metadata_key(address))
.as_ref()
.map(from_rlp)
```